src/sdc11073/mdib/containerproperties.py

```diff
@@ -456,45 +456,103 @@ def updateXMLValue(self, instance, node):
             subNode.text = value
 
 
+def _compare_extension(left: etree_.ElementBase, right: etree_.ElementBase) -> bool:
+    # xml comparison
+    try:
+        if left.tag != right.tag:  # compare expanded names
+            return False
+        if dict(left.attrib) != dict(right.attrib):  # unclear how lxml _Attrib compares
+            return False
+    except AttributeError:  # right side is not an Element type because expected attributes are missing
+        return False
+
+    # ignore comments
+    left_children = [child for child in left if not isinstance(child, etree_._Comment)]
+    right_children = [child for child in right if not isinstance(child, etree_._Comment)]
+
+    if len(left_children) != len(right_children):  # compare children count
+        return False
+    if len(left_children) == 0 and len(right_children) == 0:
+        if left.text != right.text:  # mixed content is not allowed. only compare text if there are no children
+            return False
+
+    return all(map(_compare_extension, left_children, right_children))  # compare children but keep order
+
+
+class ExtensionLocalValue(list):
+
+    compare_method = _compare_extension
+    """may be overwritten by user if a custom comparison behaviour is required"""
+
+    # def __init__(self, value):
+    #     self.value = value or list()
+
+    def __eq__(self, other):
+        try:
+            if len(self) != len(other):
+                return False
+        except TypeError:  # len of other cannot be determined
+            return False
+        for my_element, other_element in zip(self, other):
+            if not ExtensionLocalValue.compare_method(my_element, other_element):
```

Comment:
`return all(self.__class__.compare_method(left, right) for left, right in zip(self, other))`

Reply:
done